muxink/src/codec/bytesrepr.rs

```diff
@@ -0,0 +1,172 @@
+//! Bytesrepr encoding/decoding
+use std::{fmt::Debug, marker::PhantomData};
+
+use bytes::{Buf, Bytes, BytesMut};
+use casper_types::bytesrepr::{self, FromBytes, ToBytes};
+use thiserror::Error;
+
+use super::{DecodeResult, FrameDecoder, Transcoder};
+use crate::codec::DecodeResult::Failed;
+
+#[derive(Debug, Error)]
+pub enum Error {
```

Comment:
I don't think we need to reflexively use an error `enum` every time.

Does `bytesrepr::Error` implement all the necessary traits? If so, we could just use it directly without an intermediate error type. If not, we should at least make it a newtype `struct`.

Compare the `bincode` module.

Reply:
`bytesrepr::Error` doesn't implement `std::error::Error` which is required by `Transcoder::Error`

Reply:
That makes sense then. Still, how about `struct Error(bytesrepr::Error)` instead?

Reply:
Sure, changed here: e22f7be6b7200a7a083999dd68db5462a842c9ec